### PR TITLE
Functions - Run individual function while developing #214

### LINKED ISSUE #214: Functions - Run individual function while developing
From https://twitter.com/kot_2010/status/1148096321007607808

Currently, I use environment variables for this as described in https://docs.microsoft.com/en-us/azure/azure-functions/disable-function

1) Annotate every function with `[Disable("FUNCTION_{NAME}_DISABLED")]`
2) Create individual run configuration in Rider, setting al environment variables `FUNCTION_{NAME}_DISABLED` to `True`, except the one I want to run

Nicer approach would be to integrate with https://blog.jetbrains.com/dotnet/2019/03/26/run-static-methods-gutter-il-viewer-install-sdks-improvements-rider-2019-1/ and make individual functions runnable.

Could be done by adding entries to `host.json` https://docs.microsoft.com/en-us/azure/azure-functions/functions-host-json?WT.mc_id=twitter-social-antchu#functions

### PR DESCRIPTION
**Pull request by maartenba**
Fixes #214

1) Adds a field to the run configuration:

image

2) Provides a "run from gutter" action:

image

### CODE REVIEW

**Comment by maartenba** [comment]
@sdubov can you check?

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostConfigurationFactory.kt [hunk 1]
```diff
@@ -63,6 +63,7 @@ class AzureFunctionsHostConfigurationFactory(type: ConfigurationType)
                     true,
                     RunnableProjectKind.None,
                     "",
+                    "",
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostConfigurationFactory.kt]: Named arguments maybe? 😃

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsConfigurationProducer.kt [hunk 2]
```diff
@@ -74,4 +89,10 @@ class AzureFunctionsConfigurationProducer
         return FileUtil.toSystemIndependentName(item.getFile()?.path ?: "") ==
                 configurationConfiguration.parameters.projectFilePath
     }
+
+    private fun isInProject(projectFilePath: String, childPath: String?): Boolean {
+        if (childPath == null) return false
+
+        return File(childPath).startsWith(File(projectFilePath).parentFile)
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsConfigurationProducer.kt:96]: Just curious, is that for case when you are in misk files?

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsRunMarkerContributor.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            }
+
+            return null
+        }
+    }
+
+    override fun getInfo(element: PsiElement): Info? {
+        tryResolveAzureFunctionName(element) ?: return null
+
+        return Info(IconLoader.getIcon("/icons/FunctionAppRunConfiguration.svg"), ExecutorAction.getActions()) { "Run Azure Function" }
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsRunMarkerContributor.kt:67]: Move icon path to constant maybe?

**Review by sdubov — APPROVED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostConfigurationFactory.kt (hunk 1 above)
Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostConfigurationFactory.kt]: Updated.

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsConfigurationProducer.kt (hunk 2 above)
Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsConfigurationProducer.kt:96]: Yes, making sure in all cases we know whether the file is in the project path (as the original file _may_  no always be included in the MSBuild project model and exist solely on disk.

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsRunMarkerContributor.kt (hunk 3 above)
Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsRunMarkerContributor.kt:67]: Will move all icons to constants in the icons PR.

**Pull request merged**